index.js

```diff
@@ -427,16 +427,38 @@ function addAdditionalProperties (schema, externalSchema, fullSchema) {
 function refFinder (ref, schema, externalSchema) {
   // Split file from walk
   ref = ref.split('#')
+
+  const searchForId = (schema, searchedId) => {
+    let objSchema
+    const explore = (schema, searchedId) => {
+      Object.keys(schema || {}).forEach((key, i, a) => {
+        if (key === '$id' && schema[key] === searchedId) {
+          objSchema = schema
+        } else if (typeof schema[key] === 'object') {
+          explore(schema[key], searchedId)
+        }
+      })
+    }
+    explore(schema, searchedId)
+    return objSchema
+  }
+
   // If external file
   if (ref[0]) {
     schema = externalSchema[ref[0]]
   }
+
   var code = 'return schema'
   // If it has a path
   if (ref[1]) {
     var walk = ref[1].split('/')
-    for (var i = 1; i < walk.length; i++) {
-      code += `['${walk[i]}']`
+    var wl = walk.length
+    if (wl === 1) {
+      return searchForId(schema, `#${ref[1]}`)
```

Comment:
Can you put in a couple of comments explaining the if condition?

Reply:
Done, let me know if I should add more details